src/libostree/ostree-repo-commit.c

```diff
@@ -770,10 +895,10 @@ write_object (OstreeRepo         *self,
               /* Don't close the base; we'll do that later */
               g_filter_output_stream_set_close_base_stream ((GFilterOutputStream*)compressed_out_stream, FALSE);
               
-              unpacked_size = g_output_stream_splice (compressed_out_stream, file_input,
-                                                      0, cancellable, error);
-              if (unpacked_size < 0)
+              if (g_output_stream_splice (compressed_out_stream, file_input, 0, cancellable, error) < 0)
```

Comment:
I actually think this is intentional, IIRC. `g_output_stream_splice` returns a `gssize`, which will be 32 bit on i386. That limits the size of the unpacked size to 4 GB. On the other hand, `g_file_info_get_size` returns a `goffset`, which is always 64 bit. I remember @jhiesey talking to me about this issue with GOutputStream.